allensdk/brain_observatory/behavior/write_behavior_nwb/__main__.py

```diff
@@ -28,10 +28,23 @@ def write_behavior_nwb(session_data, nwb_filepath):
             os.remove(filename)
 
     try:
-        session = BehaviorSession(api=BehaviorJsonApi(session_data))
-        BehaviorNwbApi(nwb_filepath_inprogress).save(session)
-        api = BehaviorNwbApi(nwb_filepath_inprogress)
-        assert sessions_are_equal(session, BehaviorSession(api=api))
+        json_session = BehaviorSession(api=BehaviorJsonApi(session_data))
+        lims_api = BehaviorLimsApi(
+            behavior_session_id=session_data['behavior_session_id'])
+        lims_session = BehaviorSession(api=lims_api)
+
+        logging.info("Comparing a BehaviorSession created from JSON "
+                     "with a BehaviorSession created from LIMS")
+        assert sessions_are_equal(json_session, lims_session, reraise=True)
```

Comment:
@njmei I would avoid using `assert` and instead raise an Exception, maybe a custom exception called `SessionsNotEqualError`